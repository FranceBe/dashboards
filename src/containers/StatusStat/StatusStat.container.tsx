import RatioCircle from 'components/RatioCircle'
import Status from 'components/Status/Status.component'
import {
  CircleContainer,
  ColumnContainer,
  CONNECTED_COLOR,
  DISCONNECTED_COLOR,
  Label,
  LineContainer,
  Separator,
  StatusContainer,
  StatusStatContainer,
} from 'containers/StatusStat/statusStat.style'
import React from 'react'

export const StatusStat: React.FC<{ connectedLength: number; disconnectedLength: number }> = ({
  connectedLength,
  disconnectedLength,
}) => {
  const totalLength = connectedLength + disconnectedLength
  return (
    <StatusStatContainer data-testid={'status-stat'}>
      <h2>Status Ratio</h2>
      <StatusContainer>
        <ColumnContainer>
          <LineContainer>
            <Status status={'connected'} /> <Label>Connected</Label>
          </LineContainer>
          <CircleContainer>
            <RatioCircle content={`${connectedLength} / ${totalLength}`} color={CONNECTED_COLOR} />
          </CircleContainer>
        </ColumnContainer>
        <Separator />
        <ColumnContainer>
          <LineContainer>
            <Status status={'disconnected'} /> <Label>Disconnected</Label>
          </LineContainer>
          <CircleContainer>
            <RatioCircle
              content={`${disconnectedLength} / ${totalLength}`}
              color={DISCONNECTED_COLOR}
            />
          </CircleContainer>
        </ColumnContainer>
      </StatusContainer>
    </StatusStatContainer>
  )
}

export default StatusStat
