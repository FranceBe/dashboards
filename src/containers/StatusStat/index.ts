import StatusStatContainer, { StatusStat } from 'containers/StatusStat/StatusStat.container'

export { StatusStat }

export default StatusStatContainer
