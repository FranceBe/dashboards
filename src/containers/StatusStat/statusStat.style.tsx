import styled from 'styled-components'
import { elements, palette, spaces } from 'styles/variables'
export const StatusStatContainer = styled.div`
  border: 1px solid ${palette.grey_primary};
  border-radius: ${spaces.small};
  box-shadow: ${elements.box_shadow};
  min-height: 370px;
  min-width: 340px;
  h2 {
    text-align: center;
  }
`

export const StatusContainer = styled.div`
  display: flex;
  justify-content: center;
`
export const ColumnContainer = styled.div`
  display: flex;
  flex-direction: column;
  padding-right: 20px;
  padding-left: 20px;
  width: 48%;
`

export const LineContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`

export const Separator = styled.div`
  border-left: 1px solid ${palette.grey_primary};
  min-height: 290px;
  width: 1px;
`

export const Label = styled.div`
  margin-left: ${spaces.regular};
`

export const CircleContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
`

export const CONNECTED_COLOR = palette.green_primary
export const DISCONNECTED_COLOR = palette.red_primary
