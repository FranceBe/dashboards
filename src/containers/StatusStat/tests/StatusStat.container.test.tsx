import '@testing-library/jest-dom/extend-expect'

import { render, screen } from '@testing-library/react'
import StatusStat from 'containers/StatusStat'
import React from 'react'

describe('StatusStat', () => {
  it('should match snapshot', () => {
    const { container } = render(<StatusStat connectedLength={23} disconnectedLength={12} />)
    expect(container.firstChild).toMatchSnapshot()
  })
  it('should display title', () => {
    render(<StatusStat connectedLength={23} disconnectedLength={12} />)
    expect(screen.getByText('Status Ratio')).toBeInTheDocument()
  })
  it('should display sub title with status icon', () => {
    render(<StatusStat connectedLength={23} disconnectedLength={12} />)
    expect(screen.getByText('Connected')).toBeInTheDocument()
    expect(screen.getByTestId('signal-on')).toBeInTheDocument()
    expect(screen.getByText('Disconnected')).toBeInTheDocument()
    expect(screen.getByTestId('signal-off')).toBeInTheDocument()
  })
  it('should display 2 ratio circle', () => {
    render(<StatusStat connectedLength={23} disconnectedLength={12} />)
    expect(screen.getAllByTestId('ratio-circle')).toHaveLength(2)
  })
})
