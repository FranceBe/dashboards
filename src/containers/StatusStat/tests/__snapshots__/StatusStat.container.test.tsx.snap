// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusStat should match snapshot 1`] = `
<div
  class="sc-dlnjwi gyKyEL"
  data-testid="status-stat"
>
  <h2>
    Status Ratio
  </h2>
  <div
    class="sc-hKFxyN hkEvbo"
  >
    <div
      class="sc-eCApnc jTCpRP"
    >
      <div
        class="sc-jSFjdj gKHwZP"
      >
        <div
          class="sc-gtsrHT jXWUFn"
          data-testid="status-container"
          status="connected"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            data-testid="signal-on"
            focusable="false"
            title="connected"
            viewBox="0 0 24 24"
          >
            <path
              d="M2 22h20V2z"
            />
          </svg>
        </div>
         
        <div
          class="sc-iCoGMd TbJxt"
        >
          Connected
        </div>
      </div>
      <div
        class="sc-fujyAs fioZlj"
      >
        <div
          class="sc-bdnxRM ljCoAi"
          color="#00C9A3"
          data-testid="ratio-circle"
        >
          23 / 35
        </div>
      </div>
    </div>
    <div
      class="sc-gKAaRy biKmfY"
    />
    <div
      class="sc-eCApnc jTCpRP"
    >
      <div
        class="sc-jSFjdj gKHwZP"
      >
        <div
          class="sc-gtsrHT eAlTkD"
          data-testid="status-container"
          status="disconnected"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            data-testid="signal-off"
            focusable="false"
            title="disconnected"
            viewBox="0 0 24 24"
          >
            <path
              d="M21 1l-8.59 8.59L21 18.18V1zM4.77 4.5L3.5 5.77l6.36 6.36L1 21h17.73l2 2L22 21.73 4.77 4.5z"
            />
          </svg>
        </div>
         
        <div
          class="sc-iCoGMd TbJxt"
        >
          Disconnected
        </div>
      </div>
      <div
        class="sc-fujyAs fioZlj"
      >
        <div
          class="sc-bdnxRM gKTCMe"
          color="#eb4034"
          data-testid="ratio-circle"
        >
          12 / 35
        </div>
      </div>
    </div>
  </div>
</div>
`;
